src/utils/cplay.c

```diff
@@ -599,12 +602,95 @@ void get_codec_aac(FILE *file, struct compr_config *config,
 	codec->format = format;
 }
 
+static int skip_id3v2_header(FILE *file)
+{
+	char buffer[BUFSZ];
+	int ret, bytes_read;
+	uint32_t header_size;
+
+	/* we only need to parse ID3v2 header in order to skip
+	 * the whole ID3v2 tag found at the beginning of the file.
+	 *
+	 * ID3v2 header has the following structure(v2.3.0):
+	 *
+	 * 1) file identifier
+	 * 	=> 3 bytes long.
+	 * 	=> has the value of ID3 for ID3v2 tag.
+	 *
+	 * 2) version
+	 * 	=> 2 bytes long.
+	 *
+	 * 3) flags
+	 * 	=> 1 byte long.
+	 *
+	 * 4) size
+	 * 	=> 4 bytes long.
+	 * 	=> the MSB of each byte is 0 so it needs to be ignored
+	 * 	when trying to parse the size.
+	 * 	=> this field doesn't include the 10 bytes which make up
+	 * 	the ID3v2 header so a +10 needs to be added to this
+	 * 	field in order to get the correct position of the end
+	 * 	of the ID3v2 tag.
+	 */
+
+	/* for now, we only support ID3v2 header found at the beginning
+	 * of the file so we need to move file cursor to the beginning
+	 * of the file
+	 */
+	ret = fseek(file, 0, SEEK_SET);
+	if (ret < 0)
+		return ret;
+
+	/* clean buffer for read operation */
+	memset(buffer, 0, BUFSZ);
```

Comment:
It's enough to have buffer size as ID3V2_HEADER_SIZE + 1 and set the trailing zero for strncmp() as `buffer[ID3V2_HEADER_SIZE] = '\0'` here.

Also, you can replace BUFSZ with `sizeof(buffer)`.

Reply:
Thanks, should be better now. Also fixed magic number in the `strncmp().` Should be a bit clearer now.